generator/osm2type.cpp

```diff
@@ -818,6 +818,82 @@ string DeterminePathGrade(OsmElement * p)
   return {};
 }
 
+string DetermineMtbRating(OsmElement * p)
+{
+  if (!p->HasTag("highway", "cycleway") || (!p->HasTag("mtb:scale") && !p->HasTag("mtb:scale:imba") && !p->HasTag("smoothness")))
+    return {};
+
+  enum eMtbRating : int
+  {
+    none = 0,
+    easy,
+    intermediate,
+    difficult,
+    expert
+  };
+
+  string mtbscale = p->GetTag("mtb:scale");
+  string imbascale = p->GetTag("mtb:scale:imba");
+  string SmoothnessType = p->GetTag("smoothness");
+
+  static std::map<std::string, eMtbRating> mtbscaleToRatingConversion = {
+    {"0", eMtbRating::easy},
+    {"S0", eMtbRating::easy},
```

Comment:
"S" values seem to be never used, see https://taginfo.openstreetmap.org/keys/mtb%3Ascale#values